src/Common/src/CoreLib/Interop/Windows/Kernel32/Interop.FormatMessage.cs

```diff
@@ -38,50 +39,51 @@ internal static string GetMessage(int errorCode)
 
         internal static string GetMessage(IntPtr moduleHandle, int errorCode)
         {
-            var sb = new StringBuilder(InitialBufferSize);
+            Span<char> buffer = stackalloc char[InitialBufferSize];
             do
             {
-                string errorMsg;
-                if (TryGetErrorMessage(moduleHandle, errorCode, sb, out errorMsg))
+                if (TryGetErrorMessage(moduleHandle, errorCode, buffer, out string errorMsg))
                 {
                     return errorMsg;
                 }
-                else
-                {
-                    // increase the capacity of the StringBuilder.
-                    sb.Capacity *= BufferSizeIncreaseFactor;
-                }
+
+                // Increase the capacity of the buffer.
+                buffer = new char[buffer.Length * BufferSizeIncreaseFactor];
             }
-            while (sb.Capacity < MaxAllowedBufferSize);
+            while (buffer.Length < MaxAllowedBufferSize);
 
             // If you come here then a size as large as 65K is also not sufficient and so we give the generic errorMsg.
             return string.Format("Unknown error (0x{0:x})", errorCode);
         }
 
-        private static bool TryGetErrorMessage(IntPtr moduleHandle, int errorCode, StringBuilder sb, out string errorMsg)
+        private static bool TryGetErrorMessage(IntPtr moduleHandle, int errorCode, Span<char> buffer, out string errorMsg)
         {
-            errorMsg = "";
-
             int flags = FORMAT_MESSAGE_IGNORE_INSERTS | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_ARGUMENT_ARRAY;
             if (moduleHandle != IntPtr.Zero)
             {
                 flags |= FORMAT_MESSAGE_FROM_HMODULE;
             }
 
-            int result = FormatMessage(flags, moduleHandle, (uint)errorCode, 0, sb, sb.Capacity, null);
-            if (result != 0)
+            int length;
+            unsafe
```

Comment:
Seems a shame it has to use unsafe code.

Reply:
To be clear, this code is all pre-existing this change; you can see it in the file I'm deleting (all I did was `move /Y corefx\src\Common\Interop\path corefx\src\Common\CoreLib\Interop\path`.

That said, what would you recommend instead?